### ISSUE TITLE
feat(model): compose a shadow V4.1 incremental token forward

### ISSUE THREAD

**Issue by anthony-chaudhary** [anthony-chaudhary/fak#13311]
# feat(model): compose a shadow V4.1 incremental token forward

```routing
repo: fak
lane: model
paths: ["internal/model/v41_forward.go","internal/model/v41_incremental.go","internal/model/v41_incremental_test.go"]
expected_steps: 3
```

## Working spine
A bounded V4.1 leaf toward 100/30; it does not establish either rate.

## Current state
forwardV41 at v41_forward.go:1678 appends input and recomputes full history; stepV41 around :2334 calls that path. Per-layer append needs an independently checkable full-stack composition before activation.

## Why this is next
Bounds one native execution seam; throughput stays with the mission gate.

## Parent context
anthony-chaudhary/fak#12640; performance frontier anthony-chaudhary/fak#13294.

## Core through-line
Add `forwardV41Step` as a shadow/test-only composition: embed one token, carry the four mHC streams through the configured layer sequence, then run the existing head. Stage bounded per-token deltas and original slice lengths/scalars; commit all per-layer appends and step-scoped publications only after the final layer and head succeed. Roll back by truncation/restoration, never by cloning prefix-sized state. Leave the production `Step` route unchanged.

## Gold-plating boundary
No production activation, benchmark success claim, sampling change or hidden old-history recomputation.

## Worker packet
Target: DS4.1 or Qwen3.8. Own only listed paths. Three steps: independent regression; named seam; focused/package gates and land. Keep implementation/test contexts separate. If scope exceeds three implementation files, return a narrower successor.
Prerequisite plan leaf: 12. Resolve it to an issue ID in the dependency block before dispatch. Do not activate a dependent seam until its prerequisite is landed and witnessed.

## Done condition / witness
Witness: `go test ./internal/model -run 'TestV41Incremental' -count=1`
The named test is a deliverable. Confirm `go test -list` includes it; empty match fails. Run owning-package and boundary gates after the last edit. Predeclare numerical tolerances.

## Verifiable Witness
```
go test ./internal/model -run 'TestV41Incremental' -count=1
```

## Definition of done
- [ ] Several continuation steps compare full vocabulary logits and retained states with full-prefix recomputation using existing strict tolerances and fixed tokens.
- [ ] Invocation counters prove exactly NumLayers single-position layer calls per decode step, with rollback on a fault in the last layer.
- [ ] State copy/allocation counters remain independent of prefix length for equal configured windows and incomplete compressor groups.
- [ ] Focused and owning-package gates pass after the final edit; independent reviewer reads the actual call path and resolving commit.

## Acceptance gate
All checklist items plus exact witness exit 0. Software validation is [SW-VERIFIED] only; physical throughput and coherent prompt-dependent generation stay with the physical qualification owner.

## Closure binding
Commit with DCO and this issue reference, explicit-path staging, and the (fak model) trailer; no agent-created PR.

## Likely files
- internal/model/v41_forward.go
- internal/model/v41_incremental.go
- internal/model/v41_incremental_test.go

## Expected steps
3

## Value
- Centrality: Core
- P1 Context: advanced - one source-backed execution seam per packet
- P2 Net value: preserved - parity, bounded memory and admission guards remain enforced
- P3 Adaptation: N/A - no adaptive policy in this leaf
- P4 Operations: advanced - exact worker scope and a witnessed production path

## Work estimate
Estimate: 3 points

## Overall completion contribution
Contribution: 3/8 points

## Completion standard
development

## Qualification status
No measured result; physical 100/30 acceptance remains outstanding.

## Dependencies
- Start blocked by: anthony-chaudhary/fak#13309

**Comment by anthony-chaudhary** [comment]
**Prerequisite landed — this issue's blocking dependency is satisfied.**

`Start blocked by: #13309` — **#13309 is CLOSED**, landed by `cb25c4f73a` (verified ancestor of `origin/main`), which delivered the index-row publication/reader seam this composition consumes:

- `V41AttentionState.appendCompressorSourceIndex` — projects the pooled KV row through the caller's index weight exactly once per completed group and stages it as a second publication under the same source layer and the same absolute half-open input range. KV latent and index key stay distinct operands.
- `Model.v41IndexReaderStep` — the reader resolves the one committed publication closing at `pos+1`, validated against the layer's resolved plan; missing/stale/duplicated-holed/wrong-source each refuse with the typed `ErrV41ForwardStage` before any reader mutation, and the returned key is a copy.
- `projectIndex` runs exactly once per completed group and zero times for an incomplete one, so no historical compressed row is reprojected.
- Witness: `go test ./internal/model -run TestV41IndexReaderStep -count=1` -> ok, 5/5 subtests (one projection per emitted row; exact parity vs `v41CompressedRows`; incomplete group emits nothing; exact source/range identity; transactional refusals).

Per this issue's own rule — *"Do not activate a dependent seam until its prerequisite is landed and witnessed"* — the prerequisite is now landed and witnessed, so the blocking dependency is discharged.

**Remaining prerequisite before dispatch**: this issue still requires **plan leaf 12** (`forwardV41Step` composition) to be resolved to an issue ID in its dependency block. That is the one step left; the #13309 dependency no longer holds it back.

**Downstream**: #13313 is `Start blocked by: #13311`, so landing this unblocks the parity-verified incremental `Session.Step` activation line (#13311 -> #13313 -> #13346).

**Comment by anthony-chaudhary** [comment]
Landed on main as 9021dc3e1 (DCO, #13311, (fak model) trailer; explicit-path staging; no agent PR).

## Deliverable
- internal/model/v41_incremental.go — orwardV41Step: shadow/test-only one-token composition. Embeds one token, carries the four mHC streams through every configured layer via the already-landed 41LayerStep seam, then runs the existing 41Head. Commits history + step-scoped publication only after the final layer and head succeed; rolls back by truncation/restoration of per-layer scalars, never by cloning prefix-sized state. Production Step/stepV41/orwardV41 routes unchanged.
- internal/model/v41_incremental_test.go — includes the ticket's named witness TestV41Incremental (plus rollback, ring-rollback, footprint, and fail-closed siblings).

## Witness (exit 0)
```
go test ./internal/model -run 'TestV41Incremental' -count=1
ok  github.com/anthony-chaudhary/fak/internal/model
```
go test -list 'TestV41Incremental' lists 9 matching tests; go vet ./internal/model/... clean; gofmt -l clean.

## Definition of done
- [x] Continuation steps compare full-vocabulary logits + retained state against full-prefix recomputation at predeclared tol=1e-6 (observed delta 0).
- [x] Invocation counters prove exactly NumLayers single-position layer calls per step; rollback on a last-layer fault (LayersRolledBack == NumLayers).
- [x] Rollback restores each layer's 
extWindowPos, 
extCompressRow, and the one ring row it overwrote — reviewed and independently reproduced.
- [x] Focused + owning-package gates pass after the final edit; independent reviewer read the actual call path.

## Notes for the record
1. An adversarial review found a real defect in the **first** implementation: rollback restored only 
extWindowPos, but V41AttentionState.Step also overwrites a ring row at 
extWindowPos % windowSize. Once wrapped (pos >= 128), a later layer's fault corrupted an earlier layer's retained ring. Fixed by staging/restoring the overwritten ring row content and 
extCompressRow; the new TestV41IncrementalRingRollbackPreservesWrappedSlot fails pre-fix and passes now.
2. A peer concurrently landed 3fa81e0da (same two paths). This commit was rebased onto it; the resolution keeps the superset version (this ticket's exact named TestV41Incremental plus the ring-rollback fix) and is now the main tip.
3. Pre-existing unrelated red tests in internal/model (TestVerbalizableDirectionLiveArtifact missing external fixture; TestExpertResidencyLFUDecayBeatsLRUOnRoutingJitter) fail identically on a clean baseline without these two files — peer WIP, not this change.
